src/whylogs/core/model_profile.py

```diff
@@ -64,41 +67,50 @@ def compute_metrics(self, targets,
 
         """
         tgt_type = type_of_target(targets)
-        if tgt_type not in ("binary", "multiclass"):
-            raise NotImplementedError("target type not supported yet")
-        # if score are not present set them to 1.
-        if scores is None:
-            scores = np.ones(len(targets))
-
-        scores = np.array(scores)
-
-        # compute confusion_matrix
-        self.metrics.compute_confusion_matrix(predictions=predictions,
-                                              targets=targets,
-                                              scores=scores,
-                                              target_field=target_field,
-                                              prediction_field=prediction_field,
-                                              score_field=score_field)
+        if tgt_type in ("continuous") or model_type == ModelType.REGRESSION:
+
+            self.metrics.compute_regression_metrics(predictions=predictions,
+                                                    targets=targets,
+                                                    target_field=target_field,
+                                                    prediction_field=prediction_field)
+            self.metrics.model_type = ModelType.REGRESSION
+
+        elif tgt_type in ("binary", "multiclass") or model_type == ModelType.CLASSIFICATION:
+            self.metrics.model_type = ModelType.CLASSIFICATION
+
+            # if score are not present set them to 1.
+            if scores is None:
+                scores = np.ones(len(targets))
```

Comment:
np.ones() returns an array, so does it work to wrap that in np.array() a couple lines later?

Reply:
scores may be just a list from the user side.